contrib/platform/src/com/sun/jna/platform/win32/Wbemcli.java

```diff
@@ -0,0 +1,172 @@
+    public static final int WBEM_S_NO_ERROR = 0x0;
+    public static final int WBEM_S_FALSE = 0x1;
+    public static final int WBEM_S_TIMEDOUT = 0x40004;
+    public static final int WBEM_S_NO_MORE_DATA = 0x40005;
+
+    // Error constants
+    // https://docs.microsoft.com/en-us/windows/desktop/wmisdk/wmi-error-constants
+    public static final int WBEM_E_INVALID_NAMESPACE = 0x8004100e;
+    public static final int WBEM_E_INVALID_CLASS = 0x80041010;
+    public static final int WBEM_E_INVALID_QUERY = 0x80041017;
+
+    /**
+     * Holds a row of results of a WMI query
+     */
+    class IWbemClassObject extends Unknown {
+
+        public IWbemClassObject(Pointer pvInstance) {
+            super(pvInstance);
+        }
+
+        public HRESULT Get(BSTR wszName, NativeLong lFlags, VARIANT.ByReference pVal, Pointer pvtType,
+                LongByReference plFlavor) {
+            // Get is 5th method of IWbemClassObjectVtbl in WbemCli.h
+            return (HRESULT) _invokeNativeObject(4,
+                    new Object[] { getPointer(), wszName, lFlags, pVal, pvtType, plFlavor }, HRESULT.class);
+        }
+    }
+
+    /**
+     * Iterates to the next row of results of a WMI query
+     */
+    class IEnumWbemClassObject extends Unknown {
+
+        public IEnumWbemClassObject(Pointer pvInstance) {
+            super(pvInstance);
+        }
+
+        public HRESULT Next(NativeLong lTimeOut, NativeLong uCount, PointerByReference ppObjects,
+                LongByReference puReturned) {
+            // Next is 5th method of IEnumWbemClassObjectVtbl in
+            // WbemCli.h
+            return (HRESULT) _invokeNativeObject(4,
+                    new Object[] { getPointer(), lTimeOut, uCount, ppObjects, puReturned }, HRESULT.class);
+        }
+    }
+
+    /**
+     * Locates and connects to a WMI namespace
+     */
+    class IWbemLocator extends Unknown {
+        public static final CLSID CLSID_WbemLocator = new CLSID("4590f811-1d3a-11d0-891f-00aa004b2e24");
+        public static final GUID IID_IWbemLocator = new GUID("dc12a687-737f-11cf-884d-00aa004b2e24");
+
+        private IWbemLocator(Pointer pvInstance) {
+            super(pvInstance);
+        }
+
+        public static IWbemLocator create() {
+            PointerByReference pbr = new PointerByReference();
+
+            HRESULT hres = Ole32.INSTANCE.CoCreateInstance(CLSID_WbemLocator, null, WTypes.CLSCTX_INPROC_SERVER,
+                    IID_IWbemLocator, pbr);
+            if (COMUtils.FAILED(hres)) {
+                Ole32.INSTANCE.CoUninitialize();
+                throw new WbemcliException("Failed to create WbemLocator object.", hres.intValue());
+            }
+
+            return new IWbemLocator(pbr.getValue());
+        }
+
+        public HRESULT ConnectServer(BSTR strNetworkResource, BSTR strUser, BSTR strPassword, BSTR strLocale,
+                NativeLong lSecurityFlags, BSTR strAuthority, Pointer pCtx, PointerByReference ppNamespace) {
```

Comment:
You could use String instead of BSTR and allocate + release the BSTRs inside the function (see below for allocation of BSTR).

For `pCtx` should this really be a plain Pointer?

Reply:
For `pCtx` it would probably be an [IWbemContext](https://docs.microsoft.com/en-us/windows/desktop/api/WbemCli/nn-wbemcli-iwbemcontext) object.  But in all the sample code I have seen on MS sites for WMI connections (both local and remote) it's been `null`.   Reading the docs, it looks like `IUnknown` (or is there a ByReference version of it) may be correct?